sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveShim.scala

```diff
@@ -642,11 +641,11 @@ private[client] class Shim_v0_13 extends Shim_v0_12 {
           col.getType.startsWith(serdeConstants.CHAR_TYPE_NAME))
         .map(col => col.getName).toSet
 
-      def unapply(attr: Attribute): Option[String] = {
+      def unapply(attr: Attribute): Option[Attribute] = {
```

Comment:
why this change?

Reply:
Because I need a Attribute to be passed into `decoratedValue`, which will process the Literal -- add or remove quotes if needed.